source/retryable-writes/tests/unified/insertOne-serverErrors.yml

```diff
@@ -75,3 +75,123 @@ tests:
           - { _id: 1, x: 11 }
           - { _id: 2, x: 22 }
           - { _id: 3, x: 33 }  # The write was still applied
+
+  - description: "RetryableWriteError label is added based on top-level code in pre-4.4 server response"
+    runOnRequirements:
+      - minServerVersion: "4.2"
+        maxServerVersion: "4.2.99"
+        topologies: [ replicaset, sharded ]
+    operations:
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *client0
+          failPoint:
+            configureFailPoint: failCommand
+            # Trigger the fail point twice to allow asserting the error label in
+            # the retry attempt's response.
+            mode: { times: 2 }
+            data:
+              failCommands: [ "insert" ]
+              errorCode: 189 # PrimarySteppedDown
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: { _id: 3, x: 33 }
+        expectError:
+          errorLabelsContain: [ "RetryableWriteError" ]
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent: &insertCommandStartedEvent
+              command:
+                insert: *collectionName
+                documents: [{ _id: 3, x: 33 }]
+              commandName: insert
+              databaseName: *databaseName
+          - commandStartedEvent: *insertCommandStartedEvent
+    outcome:
+      - collectionName: *collectionName
+        databaseName: *databaseName
+        documents:
+          - { _id: 1, x: 11 }
+          - { _id: 2, x: 22 }
```

Comment:
In contrast to the mocked writeConcernError tests below, failing `insert` with an `errorCode` does prevent the write from being applied.